startblog/views.py

```diff
@@ -13,39 +13,65 @@
 from django.core.mail import send_mail, BadHeaderError
 from django.template.loader import render_to_string
 from django.http import HttpResponse
-from django.db.models import Exists,OuterRef,Count,Subquery
+from django.db.models import Exists, OuterRef, Count, Subquery
+from .tasks import *
+
 # Create your views here.
 
+
 def home(request):
     if not request.user.is_authenticated:
-        return redirect('blogapp:login')
-    blog = Blogs.objects.annotate ( is_liked = Exists(Likes.objects.filter(liked_user=request.user,post__id=OuterRef('pk'))),count =Count('blog_likes')).all()
-    context = {'blog':blog}
-    return render (request,'startblog/bloghome.html',context)
+        return redirect('blogapp:user_login')
+    '''annotate is used to add extra attribute data to the object'''
+    blog = Blogs.objects.annotate (is_liked = Exists(Likes.objects.filter(liked_user=request.user,post__id=OuterRef('pk'))),  
+    count =Count('blog_likes')).all()
```

Comment:
add a property for likes count in Likes Model

Reply:
Done